cloud/azure/actuators/cluster/clusteractuator.go

```diff
@@ -14,30 +14,105 @@ limitations under the License.
 package cluster
 
 import (
+	"errors"
 	"fmt"
+	"log"
+	"os"
 
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/golang/glog"
+	"github.com/joho/godotenv"
+	azureconfigv1 "github.com/platform9/azure-provider/cloud/azure/providerconfig/v1alpha1"
+	"github.com/platform9/azure-provider/cloud/azure/services"
+	"github.com/platform9/azure-provider/cloud/azure/services/network"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 	client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 )
 
 type AzureClusterClient struct {
-	clusterClient client.ClusterInterface
+	services                 *services.AzureClients
+	clusterClient            client.ClusterInterface
+	azureProviderConfigCodec *azureconfigv1.AzureProviderConfigCodec
 }
 
 type ClusterActuatorParams struct {
 	ClusterClient client.ClusterInterface
+	Services      *services.AzureClients
 }
 
 func NewClusterActuator(params ClusterActuatorParams) (*AzureClusterClient, error) {
+	_, azureProviderConfigCodec, err := azureconfigv1.NewSchemeAndCodecs()
+	if err != nil {
+		return nil, err
+	}
+	azureServicesClients, err := azureServicesClientOrDefault(params)
+	if err != nil {
+		return nil, err
+	}
+
 	return &AzureClusterClient{
-		clusterClient: params.ClusterClient,
+		services:                 azureServicesClients,
+		clusterClient:            params.ClusterClient,
+		azureProviderConfigCodec: azureProviderConfigCodec,
 	}, nil
 }
 
 func (azure *AzureClusterClient) Reconcile(cluster *clusterv1.Cluster) error {
-	return fmt.Errorf("NYI: Cluster Reconciles are not yet supported")
+	glog.Infof("Reconciling cluster %v.", cluster.Name)
+
+	clusterConfig, err := azure.azureProviderConfigCodec.ClusterProviderFromProviderConfig(cluster.Spec.ProviderConfig)
+	if err != nil {
+		return err
+	}
+	networkSGFuture, err := azure.network().CreateOrUpdateNetworkSecurityGroup(clusterConfig.ResourceGroup, "ClusterAPINSG", clusterConfig.Location)
```

Comment:
should move this to a const under `securitygroups.go` and default to that name if no SG name is provided. 

Refer to that defined constant from `machineactuator.go` as well.